server/controllers/journal/cash.js

```diff
@@ -0,0 +1,411 @@
+var q = require('q'),
```

Comment:
That's fine - I didn't have time to even touch sale discard.  You will notice that sale discard is not checked on the list.